lib/token-manager/token-manager.js

```diff
@@ -0,0 +1,119 @@
+/*
+ Copyright 2017 IBM Corp.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+const log4js = require("log4js");
+const logger = log4js.getLogger("appid-token-manager");
+const request = require("request");
+
+const constants = require('../utils/constants');
+const TokenUtil = require("../utils/token-util");
+const ServiceUtil = require('../utils/service-util');
+const PublicKeyUtil = require("../utils/public-key-util");
+
+const TOKEN_PATH = '/token';
+const CUSTOM_IDENTITY_GRANT_TYPE = 'urn:ietf:params:oauth:grant-type:jwt-bearer';
+
+function TokenManager(options) {
+	logger.debug("Initializing token manager");
+	this.serviceConfig = new ServiceUtil.loadConfig('TokenManager', [
+		constants.CLIENT_ID,
+		constants.TENANT_ID,
+		constants.SECRET,
+		constants.OAUTH_SERVER_URL
+	], options);
+	PublicKeyUtil.setPublicKeysEndpoint(this.serviceConfig.getOAuthServerUrl());
+}
+
+TokenManager.prototype.getCustomIdentityTokens = function (jwsToken, scopes = []) {
+	const clientId = this.serviceConfig.getClientId();
+	const secret = this.serviceConfig.getSecret();
+	const scope = scopes.join(' ');
+	const tokenEndpoint = this.serviceConfig.getOAuthServerUrl() + TOKEN_PATH;
+
+
+	return getTokens(tokenEndpoint, clientId, secret, CUSTOM_IDENTITY_GRANT_TYPE, scope, jwsToken)
+		.then((tokenResponse) => {
+			if (!tokenResponse) {
+				logger.error('Unable to parse token response');
+				throw Error('Unable to parse token response');
+			}
+
+			const accessToken = tokenResponse['access_token'];
+			const identityToken = tokenResponse['id_token'];
+			const tokenType = tokenResponse['token_type'];
+			const expiresIn = tokenResponse['expires_in'];
+
+			return Promise.all([
+				validateToken('access', accessToken, this.serviceConfig),
+				validateToken('identity', identityToken, this.serviceConfig)
+			]).then(() => ({
+				accessToken,
+				identityToken,
+				tokenType,
+				expiresIn
+			}));
```

Comment:
Error catching?

Reply:
It's not necessary since the developer will have to catch the error on the top level while using this API. 'catch(err) {throw error}' seems redundant.